### PR TITLE
chore(ci,tests): add workflow to run bootc e2e tests on windows

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Add nightly/on demand workflow to run bootc e2e tests on windows
### What issues does this PR fix or reference?
https://github.com/containers/podman-desktop-e2e/issues/172

### How to test this PR?
It would require adding scepcial credentials into repo. TBD

### CODE REVIEW

File: .github/workflows/bootc-e2e-nightly-windows.yaml [hunk 1]
```diff
@@ -0,0 +1,236 @@
+name: Podman Desktop BootC E2E Nightly
```
Comment (benoitf) [inline .github/workflows/bootc-e2e-nightly-windows.yaml:1]: FYI, missing the copyright headers

File: .github/workflows/bootc-e2e-nightly-windows.yaml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+      env:
+        DEFAULT_FORK: 'containers'
+        DEFAULT_BRANCH: 'main'
+        DEFAULT_EXT_TESTS: '1'
+        DEFAULT_NPM_TARGET: 'test:e2e'
+        DEFAULT_ENV_VARS: 'TEST_PODMAN_MACHINE=true'
+        DEFAULT_PODMAN_OPTIONS: 'INIT=1,START=1,ROOTFUL=1,NETWORKING=0'
+        DEFAULT_EXT_REPO_OPTIONS: 'REPO=podman-desktop-extension-bootc,FORK=containers,BRANCH=main'
+        DEFAULT_VERSION: '5.0.2'
+        DEFAULT_URL: 'https://github.com/containers/podman/releases/download/v5.0.2/podman-remote-release-windows_amd64.zip'
```
Comment (benoitf) [inline .github/workflows/bootc-e2e-nightly-windows.yaml:85]: you need to update all extensions when there is a new version of podman included in Podman Desktop ?

shouldn't be part of some data grabbed from podman desktop

**Review by benoitf — COMMENTED** [review]
it looks like the dependency to podman should be externalized to this repository$

this repo should only depend on 'Podman Desktop' thing

File: .github/workflows/bootc-e2e-nightly-windows.yaml (hunk 2 above)
Comment (odockal) [reply .github/workflows/bootc-e2e-nightly-windows.yaml:85]: That is a good idea.

File: .github/workflows/bootc-e2e-nightly-windows.yaml (hunk 1 above)
Comment (odockal) [reply .github/workflows/bootc-e2e-nightly-windows.yaml:18]: done

**Comment by odockal** [comment]
> it looks like the dependency to podman should be externalized to this repository$
> 
> this repo should only depend on 'Podman Desktop' thing

@benoitf After the review feedback I have changed the workflow to use the podman version used by podman-desktop when triggered on nightly base, and then there is an option to run workflow on demand with podman version to be choose by the user. But the param's value is static, I do think the default value can be extracted dynamically based on what is in podman-desktop. If that is what you meant?

**Review by benoitf — APPROVED** [review]

**Comment by odockal** [comment]
Great, I was not sure first. Also it came to me, we could probably make it default behavior, so it would be incorporated in the pde2e images but still it would be great to have another option to override default podman version from outside. I have filled an issue for this: https://github.com/containers/podman-desktop-e2e/issues/177.

**Pull request merged**